share/picongpu/tests/CurrentDeposition/lib/python/test/CurrentDeposition/grid_class.py

```diff
@@ -0,0 +1,207 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2023 PIConGPU contributors
+Authors: Hannes Wolf
+License: GPLv3+
+
+Intoduces a grid class. A particle with defined assignment function can do a
+move (also randomly) on the grid from one position the another one. The
+current deposition and current density vector fields can then be calculated.
+"""
+
+
+import numpy as np
+from AfuncW import NGP, CIC, TSC, PSQ, W
+import random
+
+
+class grid:
+    def __init__(self, order):
+        self.order = order
+        if order == 0:
+            self.func = NGP
+        elif order == 1:
+            self.func = CIC
+        elif order == 2:
+            self.func = TSC
+        elif order == 3:
+            self.func = PSQ
+        else:
+            print("There are just assignment functions up to order 3")
+            print("implemented yet /n")
+
+    def create_grid(self):
+        """creates grid appropiate to the order of the assignement function
+
+        returns meshgrids for each direction
+        """
+        order = self.order
+        # number of cells, that a-func fits in

```

Comment:
I do not know what you mean with `a-func`.

Reply:
its a assignment function. I adjusted it